rust/core-lib/src/run_plugin.rs

```diff
@@ -35,18 +35,58 @@ pub struct Plugin<W: Write> {
     peer: PluginPeer,
 }
 
+/// An simple edit, received from a plugin.
+#[derive(Serialize, Deserialize, Debug)]
+pub struct PluginEdit {
+    pub start: u64,
+    pub end: u64,
+    pub rev: u64,
+    pub text: String,
+    /// the edit priority determines the resolution strategy when merging this edit with other
+    /// concurrent edits. The highest priority edit will be applied last.
+    pub priority: u64,
+    /// whether the inserted text prefers to be to the right of the cursor.
+    pub after_cursor: bool,
+    /// the originator of this edit: some identifier (plugin name, 'core', etc)
+    pub author: String,
+}
+
+
+/// A response to an `update` RPC sent to a plugin.
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(untagged)]
+pub enum UpdateResponse {
+    /// An edit to the buffer.
+    Edit(PluginEdit),
+    /// An acknowledgement with no action. A response cannot be Null, so we send a uint.
+    Ack(u64),
+}
+
+
+static DEBUG_PLUGING_PATH: &'static str = "PATH_TO_PLUGIN";
```

Comment:
This looks at first like an env lookup. Add a comment suggesting that it be edited in place during development (if that is indeed the intent).

Also, typo PLUGING